libraries/AP_GPS/AP_GPS.h

```diff
@@ -419,9 +419,9 @@ class AP_GPS
 
     // state of auto-detection process, per instance
     struct detect_state {
-        uint32_t detect_started_ms;
         uint32_t last_baud_change_ms;
         uint8_t current_baud;
+        bool auto_detect_baud;
```

Comment:
You know... That's how I was reading it in my head, and never actually noticed :P